app/models/atmosphere/appliance.rb

```diff
@@ -129,15 +127,40 @@ def owned_by?(user)
       appliance_set.user_id == user.id
     end
 
+    def prepaid_until
+      # Helper method which determines how long this appliance will remain prepaid
+      if deployments.blank?
+        # Invoking this method on an appl whith 0 deployments is a nasal demons
+        # scenario: the question doesn't make sense and neither will the answer.
+        # Returning fixed time is least likely to break the client.
+        result = Time.parse('2000-01-01 12:00')
+      else
+        result = deployments.max_by { |dep| dep.prepaid_until }.prepaid_until
+      end
+      result
+    end
+
+    # This method provided for backward compatibility
+    def billing_state
+      if deployments.any? { |d| d.billing_state == 'prepaid' }
+        'prepaid'
+      else
+        'expired'
```

Comment:
Regarding symbols, we need to be careful. First - afaik ActiveRecord does not automatically translate string field values into ruby symbols, so 'prepaid' != :prepaid. We, however, use enumerize to kind of bypass that. So we can't use symbols here unless we change carefully the enumerize use in line 77.